packages/pytest-simcore/src/pytest_simcore/docker_swarm.py

```diff
@@ -178,6 +178,17 @@ def docker_swarm(
     assert _is_docker_swarm_init(docker_client) is keep_docker_up
 
 
+@retry(
+    wait=wait_fixed(0.3),
+    retry=retry_if_exception_type(AssertionError),
+    stop=stop_after_delay(30),
+)
+def _wait_for_new_task_to_be_started(service: Any, old_task_ids: set[str]) -> None:
+    service.reload()
+    new_task_ids = {t["ID"] for t in service.tasks()}
+    assert len(new_task_ids.difference(old_task_ids)) == 1

```

Comment:
But this does not guarantee that the new task finished the migration. Only that it was started. Meaning it could be ongoing after this check is done.

Reply:
1. If you check how the migration service works, it creates a file named ```migration.done``` **once** it finished the migration.
see [https://github.com/ITISFoundation/osparc-simcore/blob/1b6025206bc0ed534957fbdd633c492171c2baa2/services/migration/docker/entrypoint.sh](https://github.com/ITISFoundation/osparc-simcore/blob/1b6025206bc0ed534957fbdd633c492171c2baa2/services/migration/docker/entrypoint.sh)
2. The healthcheck of the migration service returns 200 if the file is there [https://github.com/ITISFoundation/osparc-simcore/blob/6e3ecfb2d5819a06e5a4b3092ec80c14803d1b18/services/migration/docker/healthcheck.sh](https://github.com/ITISFoundation/osparc-simcore/blob/6e3ecfb2d5819a06e5a4b3092ec80c14803d1b18/services/migration/docker/healthcheck.sh)
3. there is code checking whether the tasks are in the *running* state or not. so my 2 cents here is that when the migration service is force updated that the old task is still running and it goes through the [check](https://github.com/ITISFoundation/osparc-simcore/blob/fc8f63763a36824646d0ccedf1336d10dc9e76c1/packages/pytest-simcore/src/pytest_simcore/docker_swarm.py) too fast. we will see.